designs/mixins.md

```diff
@@ -518,27 +570,52 @@ structure A1 {
 
 @mixin
 structure A2 {
-    A: String
+    A: Integer
 }
 
-structure Invalid with
-    A1
-    A2 {}
+structure Invalid with [A1 A2] {}
+```
+
+A shape MAY use a member name that has already defined, but if it does it MUST
+target the same shape. This can be done to apply additional traits to the
+member.
+
+```
+@mixin
+structure A1 {
+    @private
+    a: String
+}
+
+@mixin
+structure A2 {
+    @required
+    a: String
+}
+
+structure Valid with [A1 A2] {}
 ```
 
 
 ### Mixins in the IDL
 
-To support mixins, `structure_statement` and `union_statement` ABNF rules
+To support mixins, shape ABNF rules
 will be updated to contain an optional `mixins` production
-that comes after the shape name and before `{`. Each shape ID referenced in
+that comes after the shape name and before any `{`. Each shape ID referenced in
 the `mixins` production MUST target a shape of the same type as the
 shape being defined and MUST be marked with the `@mixin` trait.
 
 ```
+simple_shape_statement = simple_type_name ws identifier [ws mixins]
```

Comment:
This would be a pretty big simplification for the parser I think, and it would get rid of the inability for us to easily add new keywords that start with "w" in the future.

```
simple_shape_statement = simple_type_name ws identifier [mixins]
mixins = sp "with" ws "[" 1*(ws shape_id) ws "]"
```

So that means these are ok:

```
string Foo with [Bar]
string Foo with [
    Bar
    Baz
]
```

But these are not:
```
string Foo
with [Bar]

string Foo
    with [Bar]
```